src/core/trtserver.cc

```diff
@@ -1209,6 +1210,26 @@ TRTSERVER_ServerSharedMemoryAddress(
   return nullptr;  // success
 }
 
+TRTSERVER_Error*
+TRTSERVER_ServerSharedMemoryStatus(
+    TRTSERVER_Server* server, TRTSERVER_Protobuf** status)
+{
+  ni::InferenceServer* lserver = reinterpret_cast<ni::InferenceServer*>(server);
+
+  ni::ServerStatTimerScoped timer(
+      lserver->StatusManager(),
+      ni::ServerStatTimerScoped::Kind::SHARED_MEMORY_CONTROL);
+
+  std::vector<SharedMemoryInfo*> active_shm_vector;
```

Comment:
Instead of having SharedMemoryInfo, why not just have the SharedMemoryStatus returned directly? Just copy what is done in TRTSERVER_ServerStatus

Reply:
active_shm_vector is no longer needed. Removed